src/bootstrap/doc.rs

```diff
@@ -60,12 +65,12 @@ macro_rules! book {
 // NOTE: When adding a book here, make sure to ALSO build the book by
 // adding a build step in `src/bootstrap/builder.rs`!
 book!(
-    CargoBook, "src/tools/cargo/src/doc", "cargo";
-    EditionGuide, "src/doc/edition-guide", "edition-guide";
-    EmbeddedBook, "src/doc/embedded-book", "embedded-book";
-    Nomicon, "src/doc/nomicon", "nomicon";
-    Reference, "src/doc/reference", "reference";
-    RustByExample, "src/doc/rust-by-example", "rust-by-example";
+    CargoBook, "src/tools/cargo/src/doc", "cargo", submodule = ["src", "tools", "cargo"];
+    EditionGuide, "src/doc/edition-guide", "edition-guide", submodule = ["src", "doc", "edition-guide"];
+    EmbeddedBook, "src/doc/embedded-book", "embedded-book", submodule = ["src", "doc", "embedded-book"];
+    Nomicon, "src/doc/nomicon", "nomicon", submodule = ["src", "doc", "nomicon"];
```

Comment:
Hm, I'm a little surprised that the submodules here need to be arrays and join'd together given that src/doc/nomicon seems to work OK and I think in some other places we have forward-slash paths too. Did you observe problems from that?

Reply:
No ... I guess this is premature bug avoidance, I was trying to prevent bugs on windows since I can't test there. But since other things are working fine it's probably overkill.